pkg/server/list.go

```diff
@@ -53,6 +53,14 @@ func (l *LimitedServer) list(ctx context.Context, r *etcdserverpb.RangeRequest)
 	if limit > 0 && resp.Count > r.Limit {
 		resp.More = true
 		resp.Kvs = kvs[0 : limit-1]
+
+		// count the actual number of results if there are more items in the db.
+		_, count, err := l.backend.Count(ctx, prefix)
```

Comment:
I think this might need to be modified to take a `revision` that is set by the revision returned by the list call, otherwise you might get an inconsistent result. For example, if keys in this prefix are created or deleted in between the time the list completes, and when the count call is made.